spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -46,16 +47,32 @@
 
 	private final String location;
 
-	public GoogleStorageResource(Storage storage, String location) {
+	private boolean createBlobIfNotExists = true;
```

Comment:
How about we surface this through a property rather than a private field?

Reply:
@joaoandremartins apologies for the dumb question (still building up my Java vocab): do you mean I should have a getter/setter for this or that I should be using like System.getProperty properties?

Reply:
@ChengyuanZhao no worries!
I was thinking that instead of this being in a private field, it should be in `GcpStorageProperties`. And then you would pass a `GcpStorageProperties` object to the constructor instead of a `boolean`.
WDYT?

Reply:
No, that's not. This `GoogleStorageResource` must be really "Plain Old Java Object". No any dependencies from other unrelated resources!

I'm not sure that it's OK to have a global configuration property on the matter though.

Something like `?noAutoCreate` as a parameter in the resource URL definition should be fully enough.

Reply:
If that's the case, then at lease make it a property and then call `setCreateBlobIfNotExists()` on `GoogleStorageProtocolResolver`.

I don't think GCS uses the URL to pass parameters around. Maybe we shouldn't start now.

What issues do you see with the global property?

Reply:
With the global property we have an issue that we can't control it per instance basis.
That's why I suggest something like:
```
@Value("gs://test-spring/images/spring.png?autoCreate=true|false")
private Resource remoteResource;
```
Well, and from here it even looks like we don't need that additional property at all.
We can resolve parameter already in the `GoogleStorageResource` via provided `location`.

Well, I don't say that we are going to pass parameter to the GCS, because I think it isn't possible according existing logic:
```
BlobId blobId = BlobId.of(uri.getHost(),
				uri.getPath().substring(1, uri.getPath().length()));
```

Here is a result of my testing:
![image](https://user-images.githubusercontent.com/1071590/31296755-740d2fc8-aab1-11e7-9952-187991bf4f58.png)

Pay, attention, please, to the selected row.

Reply:
`uri.getQuery().split("=")` comes to the rescue to extract the target flag.
I tend now to have it as `autoCreate=true|false` instead of just `noAutoCreate` because we may specify it as Properties Placeholder to be resolved against external properties:
```
@Value("gs://test-spring/images/spring.png?autoCreate=${my.autoCreate:false}")
```